src/date.rs

```diff
@@ -228,7 +224,7 @@ impl Display for HttpDate {
         buf[21] = b'0' + (self.min % 10) as u8;
         buf[23] = b'0' + (self.sec / 10) as u8;
         buf[24] = b'0' + (self.sec % 10) as u8;
-        f.write_str(unsafe { from_utf8_unchecked(&buf[..]) })
+        f.write_str(std::str::from_utf8(&buf[..]).unwrap())
```

Comment:
Check isn't needed. However this was a premature optimization. I think `from_utf8` is now even SIMD accelerated.

Reply:
yeah, I did the math and even if the HttpDate object is invalid, the worst that can happen is that `buf[12]` can become a `q` (`chr(ord('0') + 65535//1000)` in python). Which is *wrong*, but still valid UTF-8. So the definition of `HttpDate` would need to change for this to become unsound.

But I figure it's worth keeping the check since it doesn't show up in the benchmark as far as I can see.

Alternative solution would be to do the operations all on [`ascii`](https://docs.rs/ascii/1.0.0/ascii/) characters which gives you a free `.as_str()` from a `[AsciiChar]` But that's a dependency and you then need to deal with the unsafe in *that* dependency, so not worth it here.